WooCommerce/Classes/ViewRelated/CardPresentPayments/BuiltInCardReaderConnectionController.swift

```diff
@@ -366,6 +373,30 @@ private extension BuiltInCardReaderConnectionController {
             .store(in: &self.subscriptions)
         }
         stores.dispatch(softwareUpdateAction)
+
+
+        if featureFlagService.isFeatureFlagEnabled(.tapToPayEducation) {
+            let onboardingAction = CardPresentPaymentAction.observeBuiltInCardReaderOnboardingState { [weak self] onboardingEvents in
+                guard let self = self else { return }
+
+                onboardingEvents
+                    .subscribe(on: DispatchQueue.main)
+                    .sink { [weak self] event in
+                        guard let self = self else { return }
+                        switch event {
+                        case .didAcceptTermsOfService:
+                            self.isMerchantEducationInProgress.send(true)
+                            self.alertsPresenter.presentMerchantEducation { [weak self] in
+                                self?.isMerchantEducationInProgress.send(false)
+                            }
+                        }
+                    }
+                    .store(in: &self.subscriptions)
```

Comment:
super nit: can remove a few selves
```suggestion
                guard let self else { return }

                onboardingEvents
                    .subscribe(on: DispatchQueue.main)
                    .sink { [weak self] event in
                        guard let self else { return }
                        switch event {
                        case .didAcceptTermsOfService:
                            isMerchantEducationInProgress.send(true)
                            alertsPresenter.presentMerchantEducation { [weak self] in
                                self?.isMerchantEducationInProgress.send(false)
                            }
                        }
                    }
                    .store(in: &subscriptions)
```